pool/boltdb.go

```diff
@@ -243,29 +243,41 @@ func (db *BoltDB) fetchSoloPoolMode() (bool, error) {
 		if b == nil {
 			return dbError(ErrValueNotFound, "no pool mode found")
 		}
-		mode = binary.LittleEndian.Uint32(b)
+		dbMode = binary.LittleEndian.Uint32(b)
 		return nil
 	})
 	if err != nil {
-		return false, err
+		return "", err
```

Comment:
Need to have an unknown state as well instead of falling back on an empty string I think